lib/bulbs-elements/store/store.js

```diff
@@ -42,13 +65,15 @@ function dispatchFunction (store, action, payload) {
     console.time(`dispatch ${action.type}`);
   }
 
-  let nextState = cloneObject(store.state);
-  let actionState = nextState[action.fieldName];
-  let actionNextState = cloneObject(actionState);
-  actionNextState = action(actionNextState, payload, store);
-  nextState[action.fieldName] = actionNextState;
+  let priorState = store.state;
+  let nextState = Object.assign({}, priorState);
+  let priorFieldValue = priorState[action.fieldName];
+  let nextFieldValue = cloneObject(priorFieldValue);
```

Comment:
Yeah, I think we can self police this unless it bites us too much. We just have to keep in mind that we have minimal protection against pass-by-reference.